crates/opencascade-sys/build.rs

```diff
@@ -54,3 +64,65 @@ fn main() {
     println!("cargo:rerun-if-changed=src/lib.rs");
     println!("cargo:rerun-if-changed=include/wrapper.hxx");
 }
+
+struct OcctConfig {
+    include_dir: std::path::PathBuf,
+    library_dir: std::path::PathBuf,
+    is_dynamic: bool,
+}
+
+impl Default for OcctConfig {
+    #[cfg(feature = "builtin")]
+    fn default() -> Self {
+        let include_dir = occt_sys::occt_include_path();
+        let library_dir = occt_sys::occt_lib_path();
+
+        Self { include_dir, library_dir, is_dynamic: false }
+    }
+
+    #[cfg(not(feature = "builtin"))]
+    fn default() -> Self {
+        let dst = std::panic::catch_unwind(|| cmake::Config::new("OCCT_PKG").build())
+            .expect("Pre-installed OpenCASCADE library not found! You can use `builtin` feature if you do not want install OCCT libraries system-wide.");
```

Comment:
Supernit: I think it is customary that `Default::default()` is a cheap method. This one is heavyweight (builds the stub cmake project), so we could perhaps make these inherent methods called `fn detect() -> Self` or similar.